### ISSUE TITLE
Remove Task.Delay from delete tag test

### ISSUE THREAD

**Issue by annelo-msft** [Azure/azure-sdk-for-net#19698]
The service team has shared that delete tag should return a result only after a tag is deleted, so any subsequent GET operations should succeed.  Look into why there appears to be a race condition here with the GET arriving while the DELETE request is still in progress.

The goal of this issue is to remove the Task.Delay() from this test entirely.  See suggestions in [this PR](https://github.com/Azure/azure-sdk-for-net/pull/19680).

**Comment by annelo-msft** [comment]
Duplicate of https://github.com/Azure/azure-sdk-for-net/issues/19699
